Packs/ServiceNow/Integrations/ServiceNowEventCollector/ServiceNowEventCollector.py

```diff
@@ -131,51 +234,43 @@ def fetch_events_command(client: Client, last_run: dict):
           Audit logs from Workday.
 
     """
-    events = []
-    previous_run_ids = set(last_run.get("previous_run_ids", set()))
+    collected_events = []
+    for log_type in log_types:
+        previous_run_ids = set(last_run.get(PREVIOUS_RUN_IDS[log_type], set()))
+        from_date = initialize_from_date(last_run, log_type)
+        demisto.debug(f"Getting Audit Logs {from_date=}.")
+        new_events = client.search_events(from_date, log_type)
 
-    if "last_fetch_time" not in last_run:
-        current_time = datetime.utcnow()
-        first_fetch_time = current_time - timedelta(minutes=1)
-        first_fetch_str = first_fetch_time.strftime(LOGS_DATE_FORMAT)
-        from_date = first_fetch_str
-    else:
-        from_date = last_run.get("last_fetch_time", "")
-
-    demisto.debug(f"Getting Audit Logs {from_date=}.")
-    audit_logs = client.get_audit_logs(from_date)
+        if new_events:
+            demisto.debug(f"Got {len(new_events)} {log_type} events. Begin processing.")
+            events, previous_run_ids = process_and_filter_events(
+                events=new_events, previous_run_ids=previous_run_ids, from_date=from_date, log_type=log_type
+            )
 
-    if audit_logs:
-        demisto.debug(f"Got {len(audit_logs)} audit_logs. Begin processing.")
-        events, previous_run_ids = process_and_filter_events(
-            events=audit_logs, previous_run_ids=previous_run_ids, from_date=from_date
-        )
-
-        demisto.debug(f"Done processing {len(events)} audit_logs.")
-        last_fetch_time = events[-1].get("sys_created_on") if events else from_date
-        last_run = {
-            "last_fetch_time": last_fetch_time,
-            "previous_run_ids": list(previous_run_ids),
-        }
-        demisto.debug(f"Saving last run as {last_run}")
+            demisto.debug(f"Done processing {len(events)} {log_type} events.")
+            last_fetch_time = events[-1].get("sys_created_on") if events else from_date
+            last_run = update_last_run(last_run, log_type, last_fetch_time, list(previous_run_ids))
+            demisto.debug(f"Saving last run as {last_run}")
```

Comment:
Take this log outside the if it would be printed once for each event type and this will be confusing.
Actaully please move this log to be just one line before demisto.SetLastRun()